cumulus_library/builders/file_upload_builder.py

```diff
@@ -0,0 +1,101 @@
+"""Class for generating counts tables from templates"""
+
+import pathlib
+import re
+
+import pandas
+
+from cumulus_library import BaseTableBuilder, base_utils, errors, study_manifest
+from cumulus_library.template_sql import base_templates
+
+"""
+A file upload config looks like this
+
+type="file_upload"
+[tables]
+[table.name] # the table in db will be called 'name', snaked cased for sql compatibility
+file = "relative/path/from/config/file.csv"
+# The rest of these are optional
+delimiter = "|" # comma by default, only applies to csv/tsv/bsv
+col_types =["STRING","DATE","DOUBLE"] # default is all strings. Only relevant for Athena.
+# See https://cwiki.apache.org/confluence/display/Hive/LanguageManual+Types for valid types
+[table.other_name]
+file = "other/path/file.csv"
+"""
+
+
+class FileUploadBuilder(BaseTableBuilder):
+    display_text = "Uploading files..."
+
+    @staticmethod
+    def snake_case(x: str) -> str:
+        """Converts either CAPS_CASE or CamelCase to snake_case"""
+        return re.sub(r"([a-z])([A-Z])", r"\1_\2", x).lower()
+
+    def __init__(self, *args, workflow_config: dict, toml_config_path: pathlib.Path, **kwargs):
+        self.workflow_config = workflow_config
+        self.toml_config_path = toml_config_path
+        self.toml_config_dir = toml_config_path.parent
+        super().__init__(*args, **kwargs)
+
+    def prepare_queries(
+        self,
+        config: base_utils.StudyConfig,
+        manifest: study_manifest.StudyManifest,
+        *args,
+        **kwargs,
+    ):
+        with base_utils.get_progress_bar() as progress:
+            task = progress.add_task(
+                "Uploading static files...",
+                total=len(self.workflow_config["tables"]),
+            )
+            for table_name in self.workflow_config["tables"]:
+                table = self.workflow_config["tables"][table_name]
+                if table["file"].endswith(".xlsx"):
```

Comment:
Do you know this is needed or was this thrown in because pandas makes it easy? Because maybe it's worth holding off on features like this unless we know we need them (just thinking about like, a world where we switch away from pandas for some reason and also want to keep feature parity while doing so).